crates/core/src/router/mod.rs

```
@@ -202,9 +205,12 @@ impl Router {
 
             let mut peer_distances: Vec<_> = peers
                 .into_iter()
-                .filter_map(|peer| {
-                    peer.location
-                        .map(|loc| (peer, target_location.distance(loc)))
+                .map(|peer| {
```

all peers here should have their location set anyway no? but why default to an arbitrary distance instead of filter_map?

---

Good catch—updated the random fallback to skip peers without a location instead of inventing a distance, so we only rank real candidates.